src/ore/src/id_gen.rs

```diff
@@ -57,6 +58,37 @@ impl<Id: From<u64>> Gen<Id> {
 /// A generator of u64-bit IDs.
 pub type IdGen = Gen<u64>;
 
+/// Manages the allocation of unique IDs.
+///
+/// Atomic version of `Gen`, for sharing between threads.
+#[derive(Debug)]
+pub struct AtomicGen<Id> {
+    id: AtomicU64,
+    phantom: PhantomData<Id>,
+}
+
+impl<Id> Default for AtomicGen<Id> {
+    fn default() -> Self {
+        Self {
+            id: AtomicU64::new(0),
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<Id: From<u64> + Default> AtomicGen<Id> {
+    /// Allocates a new identifier of type `Id` and advances the generator.
+    pub fn allocate_id(&self) -> Id {
+        let id = self.id.fetch_add(1, Ordering::Relaxed);
```

Comment:
Also, do we no longer care about overflow? Seems pretty unlikely, but previously we explicitly were handling it.

Reply:
Other ID generators we have (using `IdGen`) also don't check for overflow, so I figured it's fine to skip it here as well. It's indeed extremely unlikely that we'd ever run out of transient IDs, especially considering our weekly maintenance window.

Reply:
I did the math: If we allocated 1000 IDs every second we'd need 585 million years to overflow. I think we're probably good :)